queue_services/entity-filer/tests/unit/filing_processors/test_annual_report.py

```diff
@@ -98,3 +100,34 @@ async def test_process_ar_filing_no_agm(app, session):
     assert filing.status == Filing.Status.COMPLETED.value
     assert business.last_agm_date == agm_date
     assert datetime.datetime.date(business.last_ar_date) == ar_date
+
+
+def test_process_ar_filing_involuntary_dissolution(app, session):
+    """Assert that an AR filling can be applied to the model correctly."""
+    from entity_filer.filing_processors import annual_report
+    # vars
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+    identifier = 'CP1234567'
+
+    # create a business that is not eligible for dissolution. ('CP' is not eligible types)
+    business = create_business(identifier, 'CP')
+    # create the batch and batch_processing that are in processing.
+    batch = create_batch()
+    batch_processing = create_batch_processing(business, batch.id)
+    now = datetime.date(2020, 9, 17)
+    ar_date = datetime.date(2020, 8, 5)
+    agm_date = datetime.date(2020, 7, 1)
+    ar = copy.deepcopy(ANNUAL_REPORT)
+    ar['filing']['business']['identifier'] = identifier
+    ar['filing']['annualReport']['annualReportDate'] = ar_date.isoformat()
+    ar['filing']['annualReport']['annualGeneralMeetingDate'] = agm_date.isoformat()
+
+    filing_meta = FilingMeta()
+
+    # TEST
+    with freeze_time(now):
+        filing = create_filing(payment_id, ar, business.id)
+        annual_report.process(business, filing.filing_json['filing'], filing_meta=filing_meta)
+
+    # check it out
+    assert batch_processing.status == BatchProcessing.BatchProcessingStatus.WITHDRAWN.value
```

Comment:
There are multiple scenarios that need to be tested either in this test or in a bunch of tests.


When `enable-involuntary-dissolution` = True:
* a business that is not in dissolution does is able to file an AR successfully
* a business that is in dissolution and is no longer eligible for dissolution after filing an AR, is withdrawn from the dissolution process.  I think this is the scenario you are trying to test in your current test.
* a business that is in dissolution and is still eligible for dissolution after filing an AR, is not withdrawn from the dissolution process


When `enable-involuntary-dissolution` = False:
* existing functionality still functions as expected

Reply:
@kzdev420 it sounded like you have addressed these scenarios I've mentioned here.  

I'm not seeing the changes yet.  Do you still have to commit them?

Reply:
Your latest updates from what I can see still don't address some of the scenarios I listed.

![image](https://github.com/bcgov/lear/assets/6685223/ed8a6d0d-63b1-4ee8-9573-f99af9f8a8d4)

Reply:
Added